ql/src/test/results/clientpositive/perf/tez/constraints/query54.q.out

```diff
@@ -202,156 +202,154 @@ Stage-0
                                           predicate:(_col1 <= _col3)
                                           Merge Join Operator [MERGEJOIN_294] (rows=15218525 width=12)
                                             Conds:(Inner),Output:["_col0","_col1","_col3"]
-                                          <-Reducer 15 [CUSTOM_SIMPLE_EDGE]
+                                          <-Reducer 20 [CUSTOM_SIMPLE_EDGE]
                                             PARTITION_ONLY_SHUFFLE [RS_99]
                                               Filter Operator [FIL_98] (rows=608741 width=12)
                                                 predicate:(_col2 <= _col1)
                                                 Merge Join Operator [MERGEJOIN_291] (rows=1826225 width=12)
                                                   Conds:(Inner),Output:["_col0","_col1","_col2"]
                                                 <-Map 9 [CUSTOM_SIMPLE_EDGE] vectorized
-                                                  PARTITION_ONLY_SHUFFLE [RS_327]
```

Comment:
Change of algorithm to SHUFFLE. Is this expected? It seems the same changed happened for multiple ops in the plan.

Reply:
this is hightly unfortunate:
the jsonexplain api "tells" the vertex about the outgoing edge type by calling [this method](https://github.com/apache/hive/blob/db895f374bf63b77b683574fdf678bfac91a5ac6/common/src/java/org/apache/hadoop/hive/common/jsonexplain/Vertex.java#L308) from [here](https://github.com/apache/hive/blob/db895f374bf63b77b683574fdf678bfac91a5ac6/common/src/java/org/apache/hadoop/hive/common/jsonexplain/Stage.java#L115)

since a single vertex can have multiple outgoing edges - setting the type of one-of-them is problematic - I think we may want to consider to simple remove this tagging of vertices

instead...we should consider renaming some of the edge types...like `CUSTOM_SIMPLE_EDGE` to `PARTITION_ONLY`